Mathlib/CategoryTheory/Monoidal/Linear.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+! This file was ported from Lean 3 source module category_theory.monoidal.linear
+! leanprover-community/mathlib commit 986c4d5761f938b2e1c43c01f001b6d9d88c2055
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Linear.LinearFunctor
+import Mathlib.CategoryTheory.Monoidal.Preadditive
+
+/-!
+# Linear monoidal categories
+
+A monoidal category is `MonoidalLinear R` if it is monoidal preadditive and
+tensor product of morphisms is `R`-linear in both factors.
+-/
+
+
+namespace CategoryTheory
+
+open CategoryTheory.Limits
+
+open CategoryTheory.MonoidalCategory
+
+variable (R : Type _) [Semiring R]
+
+variable (C : Type _) [Category C] [Preadditive C] [Linear R C]
+
+variable [MonoidalCategory C] [MonoidalPreadditive C]
```

Comment:
Since this `[MonoidalPreadditive C]` is not used in the definition of the class `MonoidalLinear`, could you remove this variable declaration and add it to a section where is is needed? (I guess it is still needed somewhere).

Reply:
Sure, I will have a look tomorrow!

Reply:
You have a porting note saying that you removed an argument in a theorem because it was unused. This is weird because the lean3 linter should have complained as well in the original file. So I checked the lean3 file and indeed, we can remove the MonoidalPreadditive C hypothesis when creating the class, and then the argument becomes unused in the theorem. So you did the right thing.

But now we have that variable declaration at the head of the file, which makes the reader think that it is used everywhere, but only a subset of the results actually use it. In particular, the class definition just below does not use it. So it would be better for readability to declare the variable `[MonoidalPreadditive C]` where it is actually needed.

Reply:
Actually now that I read the file docstring more carefully, it looks like we want to force the `MonoidalLinear` class to take a `MonoidalPreadditive C` argument: "A monoidal category is `MonoidalLinear R` if it is monoidal preadditive and
tensor product of morphisms is `R`-linear in both factors.".
I am not familiar with the relevant math.

Reply:
ok, I think that makes sense and gave it a try.

Reply:
ok, I think that makes sense and gave it a try.